app/views/users/show.html.erb

```diff
@@ -0,0 +1,193 @@
+                          <% if current_user.goals.include?(goal) %>
+                            <%= link_to goals_user_path(id: goal.id, user_id: @user.id), method: :delete do%>
+                              <%= render partial: 'goals/goal_in_basket' %>
+                            <% end %>
+                          <% else %>
+                            <%= link_to goals_users_path(goal_id: goal.id, user_id: @user.id), method: :post do %>
+                              <%= render partial: 'goals/goal_not_in_basket' %>
+                            <% end %>
+                          <% end %>
+                        </div>
+                      </li>
+                    <% end %>
+                  </ul>
+                </div>
+              </div>
+            </div>
+          <% end %>
+        </section>
+        <section id="finished-goals" class="hidden">
+          <% @categories.each do |category| %>
+            <div class="row">
+              <div class="panel panel-primary">
+                <div class="panel-heading">
+                  <h2><a href="/categories/<%= category.id %>"><%=category.name %></a></h2>
+                </div>
+                <div class="panel-body">
+                  <ul class="list-group">
+                    <% category.display_completed_user_goals(@user).each do |goal| %>
+                      <li class="row list-group-item">
+                        <div class="col-md-11">
+                          <a href="/goals/<%= goal.id %>"><%= goal.task %></a>
+                        </div>
+                        <div class="col-md-1">
+                          <% if current_user.goals.include?(goal) %>
+                            <%= link_to goals_user_path(id: goal.id, user_id: @user.id), method: :delete do%>
+                              <%= render partial: 'goals/goal_in_basket' %>
+                            <% end %>
+                          <% else %>
+                            <%= link_to goals_users_path(goal_id: goal.id, user_id: @user.id), method: :post do %>
+                              <%= render partial: 'goals/goal_not_in_basket' %>
+                            <% end %>
+                          <% end %>
+                        </div>
+                      </li>
+                    <% end %>
+                  </ul>
+                </div>
+              </div>
+            </div>
+          <% end %>
+        </section>
+        <section id="friends" class="hidden">
+          <%= render 'users/friends_feed' %>
+        </section>
+      <% else %>
+        <br>
+        <%= form_for Connection.new do |f| %>
+          <%= hidden_field_tag "connection[friendee_id]", "#{@user.id}" %>
+          <%= submit_tag("Add Friend") %>
+        <% end %>
+      <% end %>
+    </div>
+  <% end %>
+<% else %>
+  <div class="row" id="profile-page">
+    <h1>This is the profile page of <%= @user.username %></h1>
+  </div>
+<% end %>
+
+</div>
```

Comment:
Your display depending whether or not the current user is the profile owner really seems quite similar. You could get away with less duplication if you just have some if else statements closer in.